subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,810 @@
+
+For example, if you want to use <<sec:incremental_compile,incremental compilation>>, use a separate JVM process for the compiler and prevent compilation failures from failing the build, you can use this configuration:
+
+++++
+<sample id="javaCompilerOptions" dir="userguide/java/basic" title="Setting Java compiler options">
+    <sourcefile file="build.gradle" snippet="java-compiler-options"/>
+</sample>
+++++
+
+That's also how you can change the verbosity of the compiler, disable debug output in the byte code and configure where the compiler can find annotation processors.
+
+Two common options for the Java compiler are defined at the project level:
+
+`sourceCompatibility`::
+Defines which language version of Java your source files should be treated as.
+
+`targetCompatibility`::
+Defines the minimum JVM version your code should run on, i.e. it determines the version of byte code the compiler generates.
+
+If you need or want more than one compilation task for any reason, you can either <<sec:custom_java_source_sets,create a new source set>> or simply define a new task of type api:org.gradle.api.tasks.compile.JavaCompile[]. We look at setting up a new source set next.
+
+[[sec:java_cross_compilation]]
+==== Compiling and testing Java 6/7
+
+Gradle can only run on Java version 7 or higher. However, support for running Gradle on Java 7 has been deprecated and is scheduled to be removed in Gradle 5.0. There are two reasons for deprecating support for Java 7:
+
+* Java 7 reached link:http://www.oracle.com/technetwork/java/javase/eol-135779.html[end of life]. Therefore, Oracle ceased public availability of security fixes and upgrades for Java 7 as of April 2015.
+* Once support for Java 7 has ceased (likely with Gradle 5.0), Gradle's implementation can start to use Java 8 APIs optimized for performance and usability.
+
+Gradle still supports compiling, testing, generating Javadoc and executing applications for Java 6 and Java 7. Java 5 is not supported.
+
+To use Java 6 or Java 7, the following tasks need to be configured:
+
+* `JavaCompile` task to fork and use the correct Java home
+* `Javadoc` task to use the correct `javadoc` executable
+* `Test` and the `JavaExec` task to use the correct `java` executable.
+
+The following sample shows how the `build.gradle` needs to be adjusted. In order to be able to make the build machine-independent, the location of the old Java home and target version should be configured in `GRADLE_USER_HOME/gradle.properties` footnote:[For more details on `gradle.properties` see <<sec:gradle_configuration_properties>> ] in the user's home directory on each developer machine, as shown in the example.
+
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="javaCrossCompilation" dir="java/crossCompilation" title="Configure Java 6 build">
+                <sourcefile file="gradle.properties"/>
```

Comment:
Can you use proper indentation here?

Reply:
Sure. I'd just like to point out that that was the original formatting in the Java Plugin chapter :smile:

- [ ] Fix indentation on samples

Reply:
Absolutely, many of the existing snippet pieces in their current form are not formatted properly.